encord_agents/core/utils.py

```diff
@@ -86,35 +113,31 @@ def _guess_file_suffix(url: str, lr: LabelRowV2) -> tuple[str, str]:
         A file type and suffix that can be used to store the file.
         For example, ("image", ".jpg") or ("video", ".mp4").
     """
-    fallback_mimetype = "video/mp4" if lr.data_type == DataType.VIDEO else "image/png"
-    mimetype, _ = next(
+    fallback_mimetype = _FALLBACK_MIMETYPES.get(
+        storage_item.item_type if storage_item is not None else lr.data_type, None
+    )
+    if fallback_mimetype is None:
+        raise ValueError(f"No fallback mimetype found for data type {lr.data_type}")
+
+    mimetype = next(
         (
             t
             for t in (
-                mimetypes.guess_type(url),
-                mimetypes.guess_type(lr.data_title),
-                (fallback_mimetype, None),
+                storage_item.mime_type if storage_item is not None else None,
+                mimetypes.guess_type(url)[0],
+                mimetypes.guess_type(storage_item.name)[0] if storage_item is not None else None,
+                mimetypes.guess_type(lr.data_title)[0],
+                fallback_mimetype,
             )
-            if t[0] is not None
+            if t is not None
         )
     )
     if mimetype is None:
         raise ValueError("This should not have happened")
```

Comment:
Not great pattern ;)

This really cannot happen because the list-comp has 't is not None', so if all are None then a `StopIteration` exception will happen.

If the intention here is to to beat mypy into submission, then using a `cast()` and/or an explicit type annotation would be better.